vector/build.gradle

```diff
@@ -130,87 +21,28 @@ android {
-
         //  If set, MSC3086 asserted identity messages sent on VoIP calls will cause the call to appear in the room corresponding to the asserted identity.
         //  This *must* only be set in trusted environments.
         buildConfigField "Boolean", "handleCallAssertedIdentityEvents", "false"
-
         buildConfigField "Boolean", "enableLocationSharing", "true"
         buildConfigField "String", "mapTilerKey", "\"fU3vlMsMn4Jb6dnEIFsx\""
 
-        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
-
-        // Keep abiFilter for the universalApk
-        ndk {
-            abiFilters "armeabi-v7a", "x86", 'arm64-v8a', 'x86_64'
-        }
-
-        // Ref: https://developer.android.com/studio/build/configure-apk-splits.html
-        splits {
-            // Configures multiple APKs based on ABI.
-            abi {
-                // Enables building multiple APKs per ABI.
-                enable true
-
-                // By default all ABIs are included, so use reset() and include to specify that we only
-                // want APKs for armeabi-v7a, x86, arm64-v8a and x86_64.
-
-                // Resets the list of ABIs that Gradle should create APKs for to none.
-                reset()
-
-                // Specifies a list of ABIs that Gradle should create APKs for.
-                include "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
-
-                // Generate a universal APK that includes all ABIs, so user who install from CI tool can use this one by default.
-                universalApk true
-            }
-        }
-
-        applicationVariants.all { variant ->
-            // assign different version code for each output
-            def baseVariantVersion = variant.versionCode * 10
-            variant.outputs.each { output ->
-                def baseAbiVersionCode = project.ext.abiVersionCodes.get(output.getFilter(OutputFile.ABI))
-                // Known limitation: it does not modify the value in the BuildConfig.java generated file
-                // See https://issuetracker.google.com/issues/171133218
-                output.versionCodeOverride = baseVariantVersion + baseAbiVersionCode
-                print "ABI " + output.getFilter(OutputFile.ABI) + " \t-> VersionCode = " + output.versionCodeOverride + "\n"
-            }
-        }
-
-        // The following argument makes the Android Test Orchestrator run its
-        // "pm clear" command after each test invocation. This command ensures
-        // that the app's state is completely cleared between tests.
-        testInstrumentationRunnerArguments clearPackageData: 'true'
+        buildConfigField "String", "GIT_REVISION", "\"1\""
+        buildConfigField "String", "GIT_REVISION_DATE", "\"2\""
+        buildConfigField "String", "GIT_BRANCH_NAME", "\"3\""
+        buildConfigField "String", "BUILD_NUMBER", "\"4\""
+        buildConfigField "String", "VERSION_NAME", "\"5\""
```

Comment:
This would need to be updated I guess.

Reply:
yep, the versioning and build config will need to be lifted out of the vector module and supplied via the dagger graph, it's similar for the variant specific logic, ideally only the application module would know about the different variants and the _vector_ module would receive different configuration/implementations